### ISSUE TITLE
Failing tests, incompatible types.

### ISSUE THREAD

**Issue by nileshpatra** [go-git/go-git-fixtures#3]
Hi @mcuadros thanks for your work.
However, when running tests in the package, I get:

```
$ go test -v ./internal/tgz 
# github.com/go-git/go-git-fixtures/v4/internal/tgz [github.com/go-git/go-git-fixtures/v4/internal/tgz.test]
internal/tgz/tgz_test.go:40:11: invalid operation: path != "" (mismatched types billy.Filesystem and string)
internal/tgz/tgz_test.go:41:25: cannot use path (type billy.Filesystem) as type string in argument to os.RemoveAll
internal/tgz/tgz_test.go:92:27: cannot use path (type billy.Filesystem) as type string in argument to relativeTree
internal/tgz/tgz_test.go:102:21: cannot use path (type billy.Filesystem) as type string in argument to os.RemoveAll
FAIL	github.com/go-git/go-git-fixtures/v4/internal/tgz [build failed]
FAIL
```

There's some type incompatibility, can you please clean this up?

Thanks,
Nilesh

**Comment by pravi** [comment]
With gccgo, slightly different error messages, but still fails

```
$  go test -v ./internal/tgz 
# github.com/go-git/go-git-fixtures/v4/internal/tgz [github.com/go-git/go-git-fixtures/v4/internal/tgz.test]
internal/tgz/tgz_test.go:40:25: error: incompatible types in binary expression
   40 |                 if path != "" {
      |                         ^
internal/tgz/tgz_test.go:41:47: error: argument 1 has incompatible type (need explicit conversion)
   41 |                         if err = os.RemoveAll(path); err != nil {
      |                                               ^
internal/tgz/tgz_test.go:92:42: error: argument 1 has incompatible type (need explicit conversion)
   92 |                 obt, err := relativeTree(path)
      |                                          ^
internal/tgz/tgz_test.go:102:36: error: argument 1 has incompatible type (need explicit conversion)
  102 |                 err = os.RemoveAll(path)
      |                                    ^
FAIL	github.com/go-git/go-git-fixtures/v4/internal/tgz [build failed]
FAIL
$ go version
go version go1.16.5 gccgo (Debian 11.2.0-10) 11.2.0 linux/amd64
```

**Comment by pjbgf** [comment]
Closing this as it should be fixed by #8. If the problem persists, just drop a message and we can reopen this issue.
